tests/integration/multimodal/test_multimodal_parallel.py

```diff
@@ -80,8 +80,9 @@ def multimodal_all_types_documents():
     return docs
 
 
-def test_multimodal_all_types_parallel(multimodal_all_types_documents):
+def test_multimodal_all_types_parallel(multimodal_all_types_documents, is_callback_called):
     def validate_response(resp):
+        is_callback_called._callback_called = True
```

Comment:
This is overkill and not the right way. We should use the build in and standardized way of doing it. There are two ways. Either we use `unittest.mock.Mock` like:
```
response_mock = Mock(wraps=validate_response)
index_gt_flow.index(input_fn=multimodal_all_types_documents, output_fn=response_mock)
```
Or we add `pytest-mock` as a dependency and do:
```
def test_multimodal_all_types_parallel(multimodal_all_types_documents, mocker):
response_mock = mocker.Mock(wraps=validate_response)
index_gt_flow.index(input_fn=multimodal_all_types_documents, output_fn=response_mock)
```
I'd prefer using unittest.mock.Mock.

Reply:
I am working on it in general in another PR will try to implement ur suggestions, is there a way to have it easily included in the tests?

Reply:
It is exactly as easy as I wrote. just mock the callback function and test with `mock.assert_called()` if the callback was called.

Reply:
yes, yes. I mean something more automatic without much effort, But I think there isnot. About to change in #1391